gallery_dl/extractor/fanbox.py

```diff
@@ -0,0 +1,166 @@
+# -*- coding: utf-8 -*-
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 2 as
+# published by the Free Software Foundation.
+
+"""Extractors for https://www.fanbox.cc/"""
+
+from .common import Extractor, Message
+from .. import text
+
+
+class FanboxExtractor(Extractor):
+    """Base class for fanbox extractors"""
+    category = "fanbox"
+    root = "https://www.fanbox.cc"
+    directory_fmt = ("{category}", "{creator_id}")
+    filename_fmt = "{id}_{num}.{extension}"
+    archive_fmt = "{id}_{num}"
+    _warning = True
+
+    def items(self):
+        yield Message.Version, 1
+
+        if self._warning:
+            if "FANBOXSESSID" not in self.session.cookies:
+                self.log.warning("no 'FANBOXSESSID' cookie set")
+            FanboxExtractor._warning = False
+
+        for url, data in self.posts():
+            yield Message.Directory, data
+            yield Message.Url, url, data
+
+    def posts(self):
+        """Return all relevant post objects"""
+
+    def _pagination(self, url):
+        headers = {"Origin": self.root}
+
+        while url:
+            url = text.ensure_http_scheme(url)
+            body = self.request(url, headers=headers).json()["body"]
+            for item in body["items"]:
+                for url, data in self._get_post_data_and_urls(item["id"]):
+                    yield url, data
+
+            url = body["nextUrl"]
+
+    def _get_post_data_and_urls(self, post_id):
+        """Fetch and process post data"""
+        headers = {"Origin": self.root}
+        url = "https://api.fanbox.cc/post.info?postId="+post_id
+        pbody = self.request(url, headers=headers).json()["body"]
+
+        post = {
+            "id": pbody["id"],
+            "title": pbody["title"],
+            "fee_required": pbody["feeRequired"],
+            "published": pbody["publishedDatetime"],
```

Comment:
Parse `pbody["publishedDatetime"]` with `text.parse_datetime()` and add it as a `date` field.